libs/pika/async_cuda/include/pika/async_cuda/cuda_pool.hpp

```diff
@@ -135,10 +165,21 @@ namespace pika::cuda::experimental {
         std::shared_ptr<pool_data> data;
 
     public:
+        /// \brief Construct a pool of CUDA streams and handles.
+        ///
+        /// \param device the CUDA device used for scheduling work
+        /// \param num_normal_priority_streams_per_thread the number of normal priority streams per
+        /// thread
+        /// \param num_high_priority_streams_per_thread the number of high priority streams per
+        /// thread
+        /// \param flags flags used to construct CUDA streams
+        /// \param num_cublas_handles the number of cuBLAS handles to create for the whole pool
+        /// \param num_cusolver_handles the number of cuSOLVER handles to create for the whole pool
```

Comment:
I'll try :upside_down_face: They are "arbitrary"/decent average value that seem to not perform too badly with DLA-Future.

Reply:
I think that would be enough. It's good for people interested in pika but without any knowledge of DLA-Future to know these values are based on a production project.

Reply:
I've updated the documentation for the parameters in https://github.com/pika-org/pika/pull/1309/commits/f267193fb0721a22b235785b48fe49c8692c0f0b. I'm anticipating #1294 with the documentation here. I'm still documenting the per-thread parameters here, but I'd like to get #1294 merged first.

Do you find the warnings sufficient? We're essentially making a silent API change with #1294, and I'm worried that it's too silent (especially if one doesn't read documentation or release notes)... The main alternative I can think of is to make the parameters strong types instead of plain integers, thus requiring a explicit conversion at the callsite.

Reply:
Ok, thanks, good to hear you're not too concerned. This kind of stuff is indeed why we've stuck to 0.X and semver, so I suppose I should exercise that right... it's just an uneasy feeling knowing that everything will compile just fine, but possibly perform differently.

> too overkill for the reasons above.

Yeah, I'd like to avoid that if possible.

However, I am planning on having a warning about this in DLA-Future (and basically introducing a new configuration option), so there the change should definitely not be silent.

Reply:
I'll keep this as is (#1294 has also been merged).